### PR TITLE
Add an option to change the complex return type.

### LINKED ISSUE #433: Compatibility with Intel Fortran compiler
I try to compile a simple fortran code which using zdotc function. I found that when the code is compiled using intel fortran complier and linked with blis, the result is wrong. It is OK when using gfortran. So I am wandering whether blis is compatible with Intel fortran complier. 
Blis is compiled with gcc, with blas and cblas interface enabled. I searched the code, and found all the code is C/C++,  no fortran code (except some test code) is found. So, how fortran code can call blas function? Is blis only compatible with GNU compiler?
The following is the test code.
`
program zdotc_test

     implicit none

     complex*16 :: a(10), b(10)
     complex*16 :: c
     complex*16 :: zdotc
     integer*4  :: i

     do i = 1,10
         a(i) = cmplx(i, i)
         b(i) = cmplx(i, 0)
     end do

     c = zdotc(10, a, 1, b, 1)

     write(*,*) a
     write(*,*) b
     write(*,*) c

     end program
`

### PR DESCRIPTION
**Pull request by devinamatthews**
ifort apparently does not return complex numbers in registers as in C/C++ (or gfortran), but instead creates a "hidden" first parameter for the return value. The option --complex-return=gnu|intel has been added, as well as a guess based on a provided FC if not specified (otherwise default to gnu). This option affects the signatures of cdotc, cdotu, zdotc, and zdotu, and a single library cannot be used with both GNU and Intel Fortran compilers. Fixes #433.

### CODE REVIEW

**Comment by devinamatthews** [comment]
@fgvanzee this one too.

File: configure
Comment (isuruf) [inline configure:3050]: Can we make this default to `gnu` if not given to not break scripts building BLIS?

Comment (devinamatthews) [reply configure:3050]: @isuruf it does default to `gnu` if `FC` isn't specified explicitly. Do you mean to use `gnu` even if `FC` is set?

Comment (isuruf) [reply configure:3050]: Ah, you are right. Didn't read the code correctly. Sorry

Comment (devinamatthews) [reply configure:3050]: No prob, thanks for the extra set of eyes.

**Comment by fgvanzee** [comment]
Reviewing...

**Review by fgvanzee — APPROVED** [review]
Looks good to me. Both Travis CI and AppVeyor seem to be taking Labor Day off, so I'll leave it up to you whether you want to wait for them to wake up before merging.

**Comment by devinamatthews** [comment]
All tests passed before the last commit which is just whitespace stuff, so I'll merge.

**Pull request merged**
